internal/faustjs.org/docs/gutenberg/wp-graphql-content-blocks.mdx

```diff
@@ -0,0 +1,266 @@
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If the actual block has values for those fields, it will return them, otherwise it will return `null`.
+
+```json
+{
+	"__typename" : "CoreParagraph",
+	"name": "core/paragraph",
+	"attributes": {
+    	"content": "Hello world",
+    	"className": null
+	}
+}
+```
+
+## What about innerBlocks?
+
+In order to facilitate querying `innerBlocks` fields more efficiently you want to use `contentBlocks(flat: true)` instead of `contentBlocks`.
+By passing this argument, all the blocks available (both blocks and innerBlocks) will be returned all flattened in the same list.
+
+For example given the following HTML Content:
+
+```html
+<columns>
+	<column>
+		<p>Example paragraph in Column<p>
+	</column>
+<column>
+```
+
+It will return the following blocks:
+
+```json
+[
+	{
+	  "__typename": "CoreColumns",
+		"name": "core/columns",
+		"id": "63dbec9abcf9d",
+		"parentId": null
+	},
+	{
+	  "__typename": "CoreColumn",
+	  "name": "core/column",
+	  "id": "63dbec9abcfa6",
+      "parentId": "63dbec9abcf9d"
+	},
+	{
+	  "__typename": "CoreParagraph",
+	  "name": "core/paragraph",
+	  "id": "63dbec9abcfa9",
+      "parentId": "63dbec9abcfa6",
+	  "attributes": {
+	    "content": "Example paragraph in Column 1",
+	    "className": null
+	  }
+	}
+]
+```
+
+The `CoreColumns` contains one or more `CoreColumn` and each `CoreColumn` contains a `CoreParagraph`.
+If we were to create a query to resolve this hierarchy, we would had to use:
+
+```graphql
+contentBlocks {
+    __typename
+    name
+    ...on CoreColumns {
+		innerBlocks {
+			...on CoreColumn {
+				...on CoreParagraph {
+					attributes {
+						content
+						className
+					}
+				}
+			}
+		}
+    }
+}
+```
+
+This clearly becomes an scalability issue as we don't really know how many levels deep can we use for those types. It could be two levels or it could be 50. Using [block patterns](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-patterns/) for example posses a real issue.
+In each case we would have to copy the whole fragment list again and again. This is one of the shorfalls of `wp-graphql-gutenberg` plugin but with this plugin we have a more elegant solution.
+
+This is how you request all blocks as a flat list:
+
+```graphql
+contentBlocks(flat:true) {
+    __typename
+    name
+	id: nodeId
+	parentId
+    ... on CoreColumns {
+        attributes {
+        	className
+        }
+    }
+	... on CoreColumn {
+        attributes {
+        	className
+        }
+    }
+	...on CoreParagraph {
+		attributes {
+			content
+			className
+		}
+	}
+}
+```
+
+Given the flattened list of blocks though, how can you put it back? Well that's where you use the
+`nodeId` and `parentId` fields to assign temporal unique ids for each block.
+
+The `nodeId` field assignes a temporal unique id for a specific block and the `parentId` will
+be assigned only if the current block has a parent and it will get it's `nodeId` value.
```

Comment:
```suggestion
be assigned only if the current block has a parent. If the current block does have a parent, it will get the parent's `nodeId` value.
```